platform/api/aws/ami.go

```diff
@@ -28,22 +28,9 @@ type releaseAMIs struct {
 	}
 }
 
-func (a *API) AMI() (resolvedAMI string, resolveAMIErr error) {
-	a.opts.resolvedAMILock.Lock()
-	defer a.opts.resolvedAMILock.Unlock()
-
-	// Use cache if we can
-	if a.opts.ResolvedAMI != "" {
-		return a.opts.ResolvedAMI, nil
-	}
-
-	// If successfully resolved, cache the AMI id for next call.
-	// Decide based on the named returns
-	defer func() {
-		if resolveAMIErr == nil {
-			a.opts.ResolvedAMI = resolvedAMI
-		}
-	}()
+func (a *API) AMI() (string, error) {
+	a.amiResolutionLock.Lock()
+	defer a.amiResolutionLock.Unlock()
```

Comment:
Moving resolution to `New` is a problem because some users of `platform...aws` don't actually need to deal with AMI resolution, e.g. `plume`, and so might not be able to describe images.

I think moving it back to options is the sanest option.

I don't want to drop it entirely because I want to dodge ever doing multiple http requests when we only really need one.